ebean-core/src/main/java/io/ebeaninternal/server/deploy/BeanProperty.java

```diff
@@ -554,6 +562,18 @@ public Object readSet(DbReadContext ctx, EntityBean bean) throws SQLException {
   @SuppressWarnings("unchecked")
   public void bind(DataBind b, Object value) throws SQLException {
     scalarType.bind(b, value);
+    if (bindMaxLength != null) {
+      Object obj = b.popLastObject();
+      long length = bindMaxLength.length(dbLength, obj);
+      if (length > dbLength) {
+        b.closeInputStreams();
+        String s = String.valueOf(value); // take original bind value here.
```

Comment:
I'm still unsure, if original bind value should be taken.
- For strings, it does not matter 
- For JSON objects, you'll get the toString from the JSON object instead of the JSON itself (for Map e,g `[foo=a, bar=b]` instead of `{"foo":"a", "bar":"b"}`)
- The most important thing are files here. I'm more interested in the filename instead of some object id from a stream

Reply:
How about we go with this until someone comes up with a better plan